moveit_core/trajectory_processing/src/trajectory_tools.cpp

```diff
@@ -47,4 +48,54 @@ std::size_t trajectoryWaypointCount(const moveit_msgs::msg::RobotTrajectory& tra
 {
   return std::max(trajectory.joint_trajectory.points.size(), trajectory.multi_dof_joint_trajectory.points.size());
 }
+
+std::string toString(const robot_trajectory::RobotTrajectory& trajectory, bool print_jerk)
+{
+  const moveit::core::JointModelGroup* group = trajectory.getGroup();
+  const std::vector<int>& idx = group->getVariableIndexList();
+  unsigned int count = trajectory.getWayPointCount();
+
+  if (count == 0)
+  {
+    return "Empty trajectory.";
+  }
+
+  std::stringstream ss;
+  ss << "Trajectory has " << count << " points over " << trajectory.getWayPointDurationFromStart(count - 1)
+     << " seconds.\n";
+  for (unsigned i = 0; i < count; i++)
+  {
+    const moveit::core::RobotState& point = trajectory.getWayPoint(i);
+    ss << "  waypoint " << i << "\t";
+    ss << "time " << trajectory.getWayPointDurationFromStart(i);
+    ss << " pos ";
+    for (int index : idx)
+    {
+      ss << point.getVariablePosition(index) << " ";
+    }
+    ss << "vel ";
+    for (int index : idx)
+    {
+      ss << point.getVariableVelocity(index) << " ";
+    }
+    ss << "acc ";
+    for (int index : idx)
+    {
+      ss << point.getVariableAcceleration(index) << " ";
+    }
+    if (print_jerk && i > 0)
+    {
+      const moveit::core::RobotState& prev = trajectory.getWayPoint(i - 1);
+      ss << "jrk ";
+      double dt = trajectory.getWayPointDurationFromStart(i) - trajectory.getWayPointDurationFromStart(i - 1);
+      for (int index : idx)
+      {
+        ss << (point.getVariableAcceleration(index) - prev.getVariableAcceleration(index)) / dt << " ";
+      }
+    }
+    ss << "\n";
+  }
```

Comment:
I propose to leave jerk out and add it properly through #683.
Numerical differentiation for jerk can always be done based on the rest of this output and is quite a special case.
Otherwise It would be just as necessary to add support for numeric differentiation for velocities and accelerations. (I've had bugs in the past where visualization of numeric differentiation vs. stored values for them would have helped a lot)